pinot-common/src/main/java/org/apache/pinot/common/request/context/LiteralContext.java

```diff
@@ -136,6 +137,16 @@ public LiteralContext(Literal literal) {
         }
         _value = literal.getFieldValue().toString();
         break;
+      case BIG_DECIMAL_VALUE:
+        _type = FieldSpec.DataType.BIG_DECIMAL;
+        _value = literal.getBigDecimalValue();
```

Comment:
I think we should also store `_value` as `BigDecimal`. `byte[]` is just the on-the-wire type, but real type